src/Database/Oracle/Simple/TableInfo.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE DataKinds #-}
```

Comment:
Can we put this module into a different branch? Since this seems like the next phase of generic deriving (not just generically converting the values, but also generating the SQL as well)